From 56fcc063c34f06cf90252ed0f80a76e26a7e9cd1 Mon Sep 17 00:00:00 2001
From: Sandor Yu <R01008@freescale.com>
Date: Mon, 15 Dec 2014 16:07:01 +0800
Subject: [PATCH] MLK-9997-1 ov5640: function enhancement

-Add V4L2_FRMIVAL_TYPE_DISCRETE setting in ov5640_enum_framesizes.
-Correct pixelformat setting.

Signed-off-by: Sandor Yu <R01008@freescale.com>
---
 drivers/media/platform/mxc/subdev/ov5640.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/media/platform/mxc/subdev/ov5640.c b/drivers/media/platform/mxc/subdev/ov5640.c
index 1183141..4928960 100644
--- a/drivers/media/platform/mxc/subdev/ov5640.c
+++ b/drivers/media/platform/mxc/subdev/ov5640.c
@@ -1637,7 +1637,7 @@ static int ov5640_enum_framesizes(struct v4l2_subdev *sd,
 	if (fsize->index > ov5640_mode_MAX)
 		return -EINVAL;
 
-	fsize->pixel_format = ov5640_data.pix.pixelformat;
+	fsize->type = V4L2_FRMSIZE_TYPE_DISCRETE;
 	fsize->discrete.width =
 			max(ov5640_mode_info_data[0][fsize->index].width,
 			    ov5640_mode_info_data[1][fsize->index].width);
@@ -1840,7 +1840,7 @@ static int ov5640_probe(struct i2c_client *client,
 
 	ov5640_data.io_init = ov5640_reset;
 	ov5640_data.i2c_client = client;
-	ov5640_data.pix.pixelformat = V4L2_MBUS_FMT_YUYV8_2X8;
+	ov5640_data.pix.pixelformat = V4L2_PIX_FMT_YUYV;
 	ov5640_data.pix.width = 640;
 	ov5640_data.pix.height = 480;
 	ov5640_data.streamcap.capability = V4L2_MODE_HIGHQUALITY |
-- 
1.8.0

